Lib/asyncio/timeouts.py

```diff
@@ -0,0 +1,142 @@
+import enum
+
+from types import TracebackType
+from typing import final, Any, Dict, Optional, Type
+
+from . import events
+from . import tasks
+
+
+__all__ = (
+    "Timeout",
+    "timeout",
+    "timeout_at",
+)
+
+
+class _State(enum.Enum):
+    CREATED = "created"
+    ENTERED = "active"
+    CANCELLING = "cancelling"
+    CANCELLED = "cancelled"
+    EXITED = "exited"
+
+
+@final
+class Timeout:
+
+    def __init__(self, deadline: Optional[float]) -> None:
+        self._state = _State.CREATED
+
+        self._timeout_handler: Optional[events.TimerHandle] = None
+        self._task: Optional[tasks.Task[Any]] = None
+        self._deadline = deadline
+
+    @property
+    def deadline(self) -> Optional[float]:
```

Comment:
Should it be a getter function, mimic to other asyncio APIs?

Reply:
Timer handle has `when()` method, maybe rename `deadline` -> `when()`.
`timeout.when()` looks not worse than `timeout.deadline`.

Reply:
Yeah, `when()` sounds nice.